src/SqlSquared/Parser.purs

```diff
@@ -459,157 +388,111 @@ functionDecl parseExpr = PC.try do
 import_
   ∷ ∀ m a
   . Monad m
-  ⇒ P.ParserT (Array Token) m (Sig.SqlDeclF a)
-import_ = PC.try do
+  ⇒ P.ParserT TokenStream m (Sig.SqlDeclF a)
+import_ = asErrorMessage "import declaration" do
   _ ← keyword "import"
   s ← ident
   pure $ Sig.Import s
 
-variable
-  ∷ ∀ t m
-  . Corecursive t (Sig.SqlF EJ.EJsonF)
-  ⇒ Monad m
-  ⇒ P.ParserT (Array Token) m t
-variable = PC.try do
+variable ∷ ∀ m t. SqlParser' m t
+variable = asErrorMessage "variable" do
   operator ":"
   s ← ident <|> anyKeyword
   pure $ embed $ Sig.Vari s
 
-literal
-  ∷ ∀ t m
-  . Corecursive t (Sig.SqlF EJ.EJsonF)
-  ⇒ Monad m
-  ⇒ P.ParserT (Array Token) m t
-literal = PC.try $ token >>= case _ of
+literal ∷ ∀ m t. SqlParser' m t
+literal = PC.tryRethrow $ token >>= case _ of
   Lit (String s) → pure $ embed $ Sig.Literal $ EJ.String s
   Lit (Integer i) → pure $ embed $ Sig.Literal $ EJ.Integer i
   Lit (Decimal d) → pure $ embed $ Sig.Literal $ EJ.Decimal d
-  Kw s | s == "null" → pure $ embed $ Sig.Literal $ EJ.Null
-       | s == "true" → pure $ embed $ Sig.Literal $ EJ.Boolean true
-       | s == "false" → pure $ embed $ Sig.Literal $ EJ.Boolean false
+  Kw s
+    | s == "null" → pure $ embed $ Sig.Literal $ EJ.Null
+    | s == "true" → pure $ embed $ Sig.Literal $ EJ.Boolean true
+    | s == "false" → pure $ embed $ Sig.Literal $ EJ.Boolean false
   _ → P.fail "incorrect literal"
 
-arrayLiteral
-  ∷ ∀ t m
-  . Corecursive t (Sig.SqlF EJ.EJsonF)
-  ⇒ Monad m
-  ⇒ P.ParserT (Array Token) m t
-arrayLiteral = PC.try do
+arrayLiteral ∷ ∀ m t. SqlParser' m t
+arrayLiteral = do
   operator "["
   els ← PC.sepBy expr $ operator ","
   operator "]"
   pure $ embed $ Sig.Literal $ EJ.Array $ A.fromFoldable els
 
-mapLiteral
-  ∷ ∀ t m
-  . Corecursive t (Sig.SqlF EJ.EJsonF)
-  ⇒ Monad m
-  ⇒ P.ParserT (Array Token) m t
-mapLiteral = PC.try do
+mapLiteral ∷ ∀ m t. SqlParser' m t
+mapLiteral = do
   operator "{"
   els ← PC.sepBy pair $ operator ","
   operator "}"
   pure $ embed $ Sig.Literal $ EJ.Map $ EJ.EJsonMap $ A.fromFoldable els
 
-pair
-  ∷ ∀ t m
-  . Corecursive t (Sig.SqlF EJ.EJsonF)
-  ⇒ Monad m
-  ⇒ P.ParserT (Array Token) m (t × t)
-pair = PC.try do
-  l ← expr
-  operator ":"
+pair ∷ ∀ m t. SqlParser m t (t × t)
+pair = do
+  l ← PC.try $ expr <* operator ":"
```

Comment:
`pair` is used in `mapLiteral`, is there real reason for `PC.try` here? I.e. `{foo;bar}` doesn't make sense in next thrown error. Or it's for token position error?